rakelib/license.rake

```diff
@@ -0,0 +1,45 @@
+# we need to call exit explicity  in order to set the proper exit code, otherwise
+# most common CI systems can not know whats up with this tests.
+require_relative "default_plugins"
+
+namespace "license" do
+
+  desc "run core specs"
+  #task "core" =>  ["test:setup"] do
+  task "core" do
+    require 'yaml'
+    generate_license_information
+  end
+
+end
+
+def all_installed_gems
+  Gem::Specification.all = nil
+  all = Gem::Specification
+  Gem::Specification.reset
+  all
+end
+
+def generate_license_information
+  licenses = []
+  all_installed_gems.select {|y| y.gem_dir.include?('vendor') }.sort {|v, u| v.name  <=> u.name }.each do |x|
+    if(x.license) #ah gem has license information
+      licenses.push ({:name => x.name, :version => x.version.to_s, :license => x.license, :homepage => x.homepage, :email => x.email})
```

Comment:
I'd rather see some kind of gem_license_info(x) that returns the hash, then:
`licenses.push(gem_license_info(x))` or something similar. 

There's some code duplication in these sections (license hash is hand-coded multiple times `{:name => ...` for each use case. Maybe we should find the license information first, then separately add it to the hash?

```ruby
license_info = { :name => ... }
if x.license
  license_info[:license] = x.license
else
  license_file = Dir.glob(....)...
  license_info[:license] = File.read(license_file)
...
end
```